docs/source/serialization.mdx

```diff
@@ -50,257 +61,374 @@ Ready-made configurations include the following models:
 
-ONNXRuntime includes some transformers-specific transformations to leverage optimized operations in the graph. Below
-are some of the operators which can be enabled to speed up inference through ONNXRuntime (*see note below*):
+```python
+from typing import Mapping, OrderedDict
+from transformers.onnx import OnnxConfig
 
-- Constant folding
-- Attention Layer fusing
-- Skip connection LayerNormalization fusing
-- FastGeLU approximation
 
-Some of the optimizations performed by ONNX runtime can be hardware specific and thus lead to different performances if
-used on another machine with a different hardware configuration than the one used for exporting the model. For this
-reason, when using `convert_graph_to_onnx.py` optimizations are not enabled, ensuring the model can be easily
-exported to various hardware. Optimizations can then be enabled when loading the model through ONNX runtime for
-inference.
+class DistilBertOnnxConfig(OnnxConfig):
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("input_ids", {0: "batch", 1: "sequence"}),
+                ("attention_mask", {0: "batch", 1: "sequence"}),
+            ]
+        )
+```
 
+Every configuration object must implement the `inputs` property and return a
+mapping, where each key corresponds to an expected input, and each value
+indicates the axis of that input. For DistilBERT, we can see that two inputs are
+required: `input_ids` and `attention_mask`. These inputs have the same shape of
+`(batch_size, sequence_length)` which is why we see the same axes used in the
+configuration.
 
 <Tip>
 
-When quantization is enabled (see below), `convert_graph_to_onnx.py` script will enable optimizations on the
-model because quantization would modify the underlying graph making it impossible for ONNX runtime to do the
-optimizations afterwards.
+Notice that `inputs` property for `DistilBertOnnxConfig` returns an
+`OrderedDict`. This ensures that the inputs are matched with their relative
+position within the `PreTrainedModel.forward()` method when tracing the graph.
+We recommend using an `OrderedDict` for the `inputs` and `outputs` properties
+when implementing custom ONNX configurations.
 
 </Tip>
 
-<Tip>
+Once you have implemented an ONNX configuration, you can instantiate it by
+providing the base model's configuration as follows:
+
+```python
+from transformers import AutoConfig
 
-For more information about the optimizations enabled by ONNXRuntime, please have a look at the [ONNXRuntime Github](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/transformers).
+config = AutoConfig.from_pretrained("distilbert-base-uncased")
+onnx_config = DistilBertOnnxConfig(config)
+```
 
-</Tip>
+The resulting object has several useful properties. For example you can view the
+ONNX operator set that will be used during the export:
+
+```python
+# Returns 11
+print(onnx_config.default_onnx_opset)
```

Comment:
```suggestion
>>> print(onnx_config.default_onnx_opset)
11
```
The doctest syntax (used everywhere else in the doc ;-) ) makes it easier to show the output.